lib/system/chcks.nim

```diff
@@ -16,6 +16,9 @@ proc raiseRangeError(val: BiggestInt) {.compilerproc, noinline.} =
   else:
     sysFatal(RangeDefect, "value out of range: ", $val)
 
+proc raiseIndexError4(l1, h1, l2, h2: int) {.compilerproc, noinline.} =
```

Comment:
But the index type could be an unsigned type or `int64`. This is quite complex to do properly.

Reply:
That would only be for arrays, right?
I'll revert to the old behaviors for arrays

To handle them, maybe we could take a string and call `repr` on the values, but can't find a way to put this together (and `reprInt` doesn't handle uint anyway)